src/HotChocolate/Data/test/Data.Filters.SqlServer.Tests/FilterVisitorTestBase.cs

```diff
@@ -72,37 +70,39 @@ public virtual void Init(SqlServerResource resource)
             TEntity[] entities,
             FilterConvention? convention = null)
             where TEntity : class
-            where T : IFilterInputType
+            where T : FilterInputType<TEntity>
         {
-            convention ??= new FilterConvention(x => x.UseDefault());
+            convention ??= new FilterConvention(x => x.UseDefault().BindRuntimeType<TEntity, T>());
```

Comment:
@michaelstaib  this is currently a limitation. The types have to be registered on the convention